### DISCUSSION TITLE
How to get the real type of the form field

### DISCUSSION THREAD

**Discussion by PeterChen1997** [react-hook-form/react-hook-form#9939] [Q&A] [answered]
I met a problem that the field type is not right.

like this: https://codesandbox.io/s/react-hook-form-apply-validation-ts-forked-qq6w87?file=/src/index.tsx

![image](https://user-images.githubusercontent.com/21072589/218979395-b74f4fd8-5e57-4e1b-a24b-4280d293231c.png)

If a declare the {option} field is required, and used in a compoent, but actually I'm not given the default value, it's undefined right now.

I want to get it more accurate, I try to set the {option} to be nullable, it will destory my component type in props....

how can I get it done?

**Comment by bluebill1049** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/issues/9946
